service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -362,24 +363,75 @@ public void createReferenceResource(ReferencedResource resource)
     storeResource(resource);
   }
 
+  /**
+   * Update name, description, and/or attributes of the resource.
+   *
+   * @param name name of the resource, may be null if it does not need to be updated
+   * @param description description of the resource, may be null if it does not need to be updated
+   * @param attributes resource-type specific attributes, may be null if it does not need to be
+   *     updated .
+   */
   @WriteTransaction
   public boolean updateResource(
-      UUID workspaceId, UUID resourceId, String name, String description) {
-    if (name == null && description == null) {
+      UUID workspaceId,
+      UUID resourceId,
+      @Nullable String name,
+      @Nullable String description,
+      @Nullable String attributes) {
+    if (name == null && description == null && attributes == null) {
       return false;
     }
 
     var params = new MapSqlParameterSource();
 
-    if (name != null) {
+    if (!StringUtils.isEmpty(name)) {
       params.addValue("name", name);
     }
-
-    if (description != null) {
+    if (!StringUtils.isEmpty(description)) {
       params.addValue("description", description);
     }
+    if (!StringUtils.isEmpty(attributes)) {
+      params.addValue("attributes", attributes);
+    }
+
+    StringBuilder sb = new StringBuilder("UPDATE resource SET ");
+
+    if (params.hasValue("attributes")) {
+      sb.append(DbUtils.setColumnsClause(params, "attributes"));
+    } else {
+      sb.append(DbUtils.setColumnsClause(params));
+    }
+
+    sb.append(" WHERE workspace_id = :workspace_id AND resource_id = :resource_id");
+
+    MapSqlParameterSource queryParams = new MapSqlParameterSource();
```

Comment:
I think you can just continue to append values to params here.

Reply:
good catch.